### PR TITLE
Esad/fix make check features

### LINKED ISSUE #196: Bug: make check-features
## Summary

It's a low-priority issue.

```
    Checking bitcoin-da v0.1.0 (/home/kpp/chainway/secret-sovereign-sdk/adapters/bitcoin)
error[E0432]: unresolved import `sov_rollup_interface::services::da::DaService`
  --> adapters/bitcoin/src/spec/header_stream.rs:9:5
   |
9  | use sov_rollup_interface::services::da::DaService;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `DaService` in `node::services::da`
   |
note: found an item that was configured out
  --> /home/kpp/chainway/secret-sovereign-sdk/rollup-interface/src/node/services/da.rs:20:11
   |
20 | pub trait DaService: Send + Sync + 'static {
   |           ^^^^^^^^^
   = note: the item is gated behind the `native` feature

error[E0432]: unresolved import `crate::service`
  --> adapters/bitcoin/src/spec/header_stream.rs:11:12
   |
11 | use crate::service::BitcoinService;
   |            ^^^^^^^ could not find `service` in the crate root

error[E0432]: unresolved import `pin_project`
 --> adapters/bitcoin/src/spec/header_stream.rs:7:5
  |
7 | use pin_project::pin_project;
  |     ^^^^^^^^^^^ use of undeclared crate or module `pin_project`

error: cannot determine resolution for the attribute macro `pin_project`
  --> adapters/bitcoin/src/spec/header_stream.rs:13:3
   |
13 | #[pin_project]
   |   ^^^^^^^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `pin` in this scope
  --> adapters/bitcoin/src/spec/header_stream.rs:15:7
   |
15 |     #[pin]
   |       ^^^

error[E0433]: failed to resolve: use of undeclared crate or module `tokio`
  --> adapters/bitcoin/src/spec/header_stream.rs:18:12
   |
18 |     timer: tokio::time::Interval,
   |            ^^^^^ use of undeclared crate or module `tokio`

error[E0433]: failed to resolve: use of undeclared crate or module `tokio`
  --> adapters/bitcoin/src/spec/header_stream.rs:26:20
   |
26 |             timer: tokio::time::interval(interval),
   |                    ^^^^^ use of undeclared crate or module `tokio`

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `bitcoin-da` (lib) due to 7 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `bitcoin-da` (lib test) due to 7 previous errors
error: process didn't exit successfully: `/home/kpp/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo check --all-targets --manifest-path adapters/bitcoin/Cargo.toml --no-default-features` (exit status: 101)
make: *** [Makefile:46: check-features] Error 1
```

## Steps to reproduce

1. `$ make check-features`

## Expected Result

The project is built successfully.

## Actual Result

Fail to build

### PR DESCRIPTION
**Pull request by eyusufatik**
fixes #196 

Errors regarding bitcoin da and mock da are gone.

However, there is also this:

```sh
$ make check-features

info: running `cargo check --all-targets` on citrea-fullnode (84/184)
    Checking sov-db v0.4.0-rc.3 (/Users/esad/chainway-workspace/citrea/crates/sovereign-sdk/full-node/db/sov-db)
    Checking reth-rpc-types v0.2.0-beta.7 (https://github.com/paradigmxyz/reth?tag=v0.2.0-beta.7#a44e0857)
error[E0432]: unresolved import `jsonrpsee_types`
 --> /Users/esad/.cargo/git/checkouts/reth-36d3ea1d1152b20c/a44e085/crates/rpc/rpc-types/src/eth/error.rs:3:5
  |
3 | use jsonrpsee_types::ErrorObject;
  |     ^^^^^^^^^^^^^^^ use of undeclared crate or module `jsonrpsee_types`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `reth-rpc-types` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
``

### CODE REVIEW

File: crates/bitcoin-da/src/helpers/compression.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+use std::io::Write;
+
+#[cfg(feature = "native")]
```
Comment (kpp) [inline crates/bitcoin-da/src/helpers/compression.rs:3]: Actually you may put a whole module behind cfg feature=native, so we don't even have compression module on zkvm.

File: crates/sovereign-sdk/adapters/mock-da/src/service.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@ use sov_rollup_interface::services::da::{BlobWithNotifier, DaService, SlotData};
 use tokio::sync::mpsc::{unbounded_channel, UnboundedSender};
 use tokio::sync::{broadcast, Mutex as AsyncMutex, MutexGuard as AsyncMutexGuard};
 use tokio::time;
-use tracing::{instrument, Instrument};
+use tracing::instrument::Instrument;
```
Comment (kpp) [inline crates/sovereign-sdk/adapters/mock-da/src/service.rs:16]: why?

File: crates/bitcoin-da/src/helpers/compression.rs (hunk 1 above)
Comment (eyusufatik) [reply crates/bitcoin-da/src/helpers/compression.rs:3]: da verifier uses `decompress_blob`

File: crates/sovereign-sdk/adapters/mock-da/src/service.rs (hunk 2 above)
Comment (eyusufatik) [reply crates/sovereign-sdk/adapters/mock-da/src/service.rs:16]: idk why but failed to compile all the time.

there is both an instrument attribute and a module

**Review by rakanalh — APPROVED** [review]
LGTM

File: crates/evm/src/evm/mod.rs [hunk 3]
```diff
@@ -20,13 +20,13 @@ pub(crate) mod call;
 #[cfg(feature = "native")]
 pub(crate) mod error;
 
-#[cfg(test)]
+#[cfg(all(test, feature = "native"))]
```
Comment (eyusufatik) [inline crates/evm/src/evm/mod.rs:23]: i see many of this cfg() changes.

I don't understand why some tests module is under native feature flag.

Comment (kpp) [reply crates/evm/src/evm/mod.rs:23]: Because they test mods that are behind "native" flag.

Comment (eyusufatik) [reply crates/evm/src/evm/mod.rs:23]: hmm, 

what about listing the crate itself with native flag under `[dev-dependencies]`

Comment (kpp) [reply crates/evm/src/evm/mod.rs:23]: What for? We test with all features, so those tests will still be tested. That's a fix for `make check-features` only

Comment (kpp) [reply crates/evm/src/evm/mod.rs:23]: A proper way to test it would be to move tests into that module instead of keeping `test_MODULE.rs` modules.

**Pull request merged**